src/cryptography/x509/extensions.py

```diff
@@ -47,7 +48,20 @@ def _key_identifier_from_public_key(public_key):
             serialization.PublicFormat.SubjectPublicKeyInfo
         )
 
-        data = bytes(PublicKeyInfo.load(serialized)['public_key'])
+        public_key_info = DERReader(serialized).read_single_element(SEQUENCE)
+        algorithm = public_key_info.read_element(SEQUENCE)
+        public_key = public_key_info.read_element(BIT_STRING)
+        public_key_info.check_empty()
+
+        # Double-check the algorithm structure.
+        algorithm.read_element(OBJECT_IDENTIFIER)
```

Comment:
Should we `algorithm.check_empty()`?

Reply:
AlgorithmIdentifiers end with an OPTIONAL parameters element. I've added some code to parse that down a later and then check_empty.